### ISSUE TITLE
Support Gitpods for multiple repos

### ISSUE THREAD

**Issue by atomicpirate** [gitpod-io/gitpod#7447]
## Is your feature request related to a problem? Please describe
There seems to be a strong link between a Gitpod environment and a single Git repository. But say I have a situation where I have a library in repo A, and an application that uses that library in repo B. If I'm working on a change to the library and application in tandem, I may reasonably want to have a environment that allows me to checkout and change code in both repo A and B at once. Or generalizing a bit more, it would be nice to have an environment from N repos instead of just one.

## Describe the behaviour you'd like
We need a way to describe N repos that Gitpod should launch into the new environment, whereas now the assumption is that you are working solely from one repo (the current one).

In [Start Fresh](https://github.com/gitpod-io/roadmap/issues/28), Gitpod is pondering the idea of provisioning an environment not based on any existing repository. This issue has a similar requirement: we need a file that lives outside of any repo that describes a new environment, but instead of not being connected with any repo as in Start Fresh, it would connect to N repos.

One nice integration would be with the Jira Development panel, which allows one to [see all topic branches connected with an issue](https://support.atlassian.com/jira-software-cloud/docs/view-development-information-for-an-issue/#See-repository-branches). Being able to launch a Gitpod environment that has all such repos/branches checked out and ready to go would be great.

## Describe alternatives you've considered
Yes, you could "git clone" additional repos in your .gitpod.yml file, but since this file is checked into version control with the rest of the repo it's not really the right model. If I have 28 dependencies, I don't necessarily want to clone all of those all the time. But at various times, I may need to clone different subsets of them into a Gitpod environment.

More generally, we should probably decouple the Gitpod environment a bit more from any given repository. If I am working on N repos at once, it's not clear that any of them should be the "primary" repo—they are all equal peers, so probably the description of the Gitpod environment that includes all of them doesn't really belong in any particular repo.

## Additional context
This request is similar to a couple items already in progress. One can imagine Gitpod covering three cases for the number of Git repos involved: zero, one, or N.

The zero repository environment version of Gitpod is covered by [Start Fresh](https://github.com/gitpod-io/roadmap/issues/28).

The case of a single "small" repository (with "small" meaning a tight scope, like a single microservice) is of course already well supported by Gitpod. However a special case of one repository is the monorepo—a single repo containing many projects. There is an issue requesting support for that case on #7236.

This issue is for the case of N "small" repositories.

**Comment by svenefftinge** [comment]
Thanks for the detailed write-up! 🧡 
I have a few questions that I think are hard but important to answer:
How would you as a user tell Gitpod which of the 28 repos should be cloned and which should come as a library? 
How would Gitpod know how to integrate a project based on source vs. as a packaged library? (This usually requires programming language specific configuration)

### Another not mentioned alternative
Start mutliple workspaces in parallel. For network-based dependencies (i.e. services) one would need to span a network between those workspaces. For file-based linking uploading or network mounting could be tried. On the positive side, this would allow keeping dev env configuration specific to projects.

**Comment by axonasif** [comment]
Might be related: https://github.com/gitpod-io/gitpod/issues/7249

**Comment by loujaybee** [comment]
Related: https://github.com/gitpod-io/gitpod/issues/7608
